app/models/ability.rb

```diff
@@ -0,0 +1,10 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, Post, :all
+    can(%i[create destroy], Post, user:)
+    can(:destroy, Comment, user:)
+    can :manage, :all if user.role == 'admin'
+  end
+end
```

Comment:
- [ ] a user can delete all users' posts if they have the role `admin` when I add role admin to the user I'm unable to delete all the posts so please make sure to fix that